src/PKSim.Core/Mappers/ExpressionProfileToExpressionProfileBuildingBlockMapper.cs

```diff
@@ -1,52 +1,34 @@
-﻿using OSPSuite.Core;
+﻿using System.Collections.Generic;
+using OSPSuite.Core;
 using OSPSuite.Core.Domain;
 using OSPSuite.Core.Domain.Builder;
-using OSPSuite.Core.Domain.Formulas;
 using OSPSuite.Core.Domain.Services;
-using OSPSuite.Utility;
 using PKSim.Assets;
 using PKSim.Core.Model;
-using IFormulaFactory = PKSim.Core.Model.IFormulaFactory;
 
 namespace PKSim.Core.Mappers
 {
-   public interface IExpressionProfileToExpressionProfileBuildingBlockMapper : IMapper<ExpressionProfile, ExpressionProfileBuildingBlock>
+   public interface IExpressionProfileToExpressionProfileBuildingBlockMapper : IPathAndValueBuildingBlockMapper<ExpressionProfile, ExpressionProfileBuildingBlock>
    {
    }
 
-   public class ExpressionProfileToExpressionProfileBuildingBlockMapper : IExpressionProfileToExpressionProfileBuildingBlockMapper
+   public class ExpressionProfileToExpressionProfileBuildingBlockMapper : PathAndValueBuildingBlockMapper<ExpressionProfile, ExpressionProfileBuildingBlock, ExpressionParameter>, IExpressionProfileToExpressionProfileBuildingBlockMapper
    {
-      private readonly IObjectBaseFactory _objectBaseFactory;
-      private readonly IEntityPathResolver _entityPathResolver;
-      private readonly IFormulaFactory _formulaFactory;
-      private readonly IApplicationConfiguration _applicationConfiguration;
-
-      public ExpressionProfileToExpressionProfileBuildingBlockMapper(IObjectBaseFactory objectBaseFactory, IEntityPathResolver entityPathResolver, IFormulaFactory formulaFactory, IApplicationConfiguration applicationConfiguration)
+      public ExpressionProfileToExpressionProfileBuildingBlockMapper(IObjectBaseFactory objectBaseFactory, IEntityPathResolver entityPathResolver, IApplicationConfiguration applicationConfiguration, ILazyLoadTask lazyLoadTask) :
+         base(objectBaseFactory, entityPathResolver, applicationConfiguration, lazyLoadTask)
       {
-         _objectBaseFactory = objectBaseFactory;
-         _entityPathResolver = entityPathResolver;
-         _formulaFactory = formulaFactory;
-         _applicationConfiguration = applicationConfiguration;
       }
 
-      public ExpressionProfileBuildingBlock MapFrom(ExpressionProfile expressionProfile)
+      protected override IReadOnlyList<IParameter> AllParametersFor(ExpressionProfile sourcePKSimBuildingBlock)
```

Comment:
rename the parameter here. No need to keep using the generic name of the base class IMO